src/python_testing/TC_AVSM_2_1.py

```diff
@@ -292,145 +315,174 @@ async def test_TC_AVSM_2_1(self):
 
         self.step(26)
         if self.pics_guard(self.check_pics("AVSM.S.A0018")):
             value = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr.Viewport)
             logger.info(f"Rx'd Viewport: {value}")
+            asserts.assert_is_not_none(value, "Viewport is None")
 
         self.step(27)
         if self.pics_guard(self.check_pics("AVSM.S.A0019")):
             value = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr.SpeakerMuted)
             logger.info(f"Rx'd SpeakerMuted: {value}")
+            asserts.assert_is_not_none(value, "SpeakerMuted is None")
 
         self.step(28)
         if self.pics_guard(self.check_pics("AVSM.S.A001A")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.SpeakerVolumeLevel
             )
             logger.info(f"Rx'd SpeakerVolumeLevel: {value}")
+            asserts.assert_is_not_none(value, "SpeakerVolumeLevel is None")
+            matter_asserts.assert_valid_uint8(value, "SpeakerVolumeLevel")
 
         self.step(29)
         if self.pics_guard(self.check_pics("AVSM.S.A001B")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.SpeakerMaxLevel
             )
             logger.info(f"Rx'd SpeakerMaxLevel: {value}")
+            asserts.assert_is_not_none(value, "SpeakerMaxLevel is None")
+            matter_asserts.assert_valid_uint8(value, "SpeakerMaxLevel")
 
         self.step(30)
         if self.pics_guard(self.check_pics("AVSM.S.A001C")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.SpeakerMinLevel
             )
             logger.info(f"Rx'd SpeakerMinLevel: {value}")
+            asserts.assert_is_not_none(value, "SpeakerMinLevel is None")
+            matter_asserts.assert_valid_uint8(value, "SpeakerMinLevel")
 
         self.step(31)
         if self.pics_guard(self.check_pics("AVSM.S.A001D")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.MicrophoneMuted
             )
             logger.info(f"Rx'd MicrophoneMuted: {value}")
+            asserts.assert_is_not_none(value, "MicrophoneMuted is None")
 
         self.step(32)
         if self.pics_guard(self.check_pics("AVSM.S.A001E")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.MicrophoneVolumeLevel
             )
             logger.info(f"Rx'd MicrophoneVolumeLevel: {value}")
+            asserts.assert_is_not_none(value, "MicrophoneVolumeLevel is None")
+            matter_asserts.assert_valid_uint8(value, "MicrophoneVolumeLevel")
```

Comment:
This needs to also test that it is between MicrophoneMinLevel and MicrophoneMaxLevel

Reply:
Thank you for review. This requires reading MicrophoneMinLevel and MicrophoneMaxLevel before MicrophoneVolumeLevel.

Reply:
Thank you for review. you can raise new PR.